blocks.go

```diff
@@ -68,6 +66,10 @@ func (b *blocks) FinalizedHeader() (*flowgo.Header, error) {
 }
 
 func (b *blocks) ByHeightFrom(height uint64, header *flowgo.Header) (*flowgo.Header, error) {
-	return environment.NewBlockFinder(b.headers).
-		ByHeightFrom(height, header)
+	block, err := b.blockchain.storage.BlockByHeight(context.Background(), height)
+	if err != nil {
+		return nil, err
+	}
+
+	return block.Header, nil
```

Comment:
Was there something wrong with the old approach?

Reply:
this is related to #280